suite-native/discovery/src/discoveryThunks.ts

```diff
@@ -270,13 +393,18 @@ export const createDescriptorPreloadedDiscoveryThunk = createThunk(
         const supportedNetworksSymbols = supportedNetworks.map(network => network.symbol);
         const discoveryNetworksTotalCount = supportedNetworksSymbols.length;
 
-        let availableCardanoDerivationsResponse;
+        let availableCardanoDerivations: ('normal' | 'legacy' | 'ledger')[] | undefined;
         if (supportedNetworks.some(network => network.networkType === 'cardano')) {
-            availableCardanoDerivationsResponse = await requestDeviceAccess(() =>
-                dispatch(getAvailableCardanoDerivationsThunk({ deviceState, device })).unwrap(),
-            );
+            const availableCardanoDerivationsResult = await dispatch(
+                getCardanoSupportedAccountTypesThunk({
+                    deviceState,
+                    device,
+                }),
+            ).unwrap();
 
-            if (!availableCardanoDerivationsResponse.success) return false;
+            if (availableCardanoDerivationsResult.success) {
+                availableCardanoDerivations = availableCardanoDerivationsResult.payload;
+            }
```

Comment:
preferably, this whole thing could be in the thunk you created so that you can just store the result of the thunk into the variable and use it as param below

Reply:
044d93041c